src/Components/Components/src/Forms/EditForm.cs

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Components.RenderTree;
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// Renders a form element that cascades an <see cref="EditContext"/> to descendants.
+    /// </summary>
+    public class EditForm : ComponentBase
```

Comment:
Is this useful/meaningful to subclass?

Reply:
I recognize that inheritance is a very sketchy way of extending or reusing functionality in general, but it is pretty convenient and well-performing in the case of these forms-and-input components. You can subclass `EditForm` to change how it renders (e.g., to put your own special attributes on the `<form>` tag). It's more useful still in the case of the `Input*` components - the base classes are designed around the idea that you'd subclass them to customize rendering.